lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -480,9 +480,14 @@ sub prepare_job_results {
     # prefetch test suite names from job settings
     my $job_settings
       = $self->schema->resultset('JobSettings')
-      ->search({job_id => {-in => [map { $_->id } @$jobs]}, key => 'TEST_SUITE_NAME'});
-    my %test_suite_name_by_job_id = map { $_->job_id => $_->value } $job_settings->all;
-    my %test_suite_names          = map { $_->id     => ($test_suite_name_by_job_id{$_->id} // $_->TEST) } @$jobs;
+      ->search({job_id => [map { $_->id } @$jobs], key => [qw(TEST_SUITE_DESCRIPTION TEST_SUITE_NAME)]});
+    my %settings_by_job_id;
+    for my $js ($job_settings->all) {
+        $settings_by_job_id{$js->key}->{$js->job_id} = $js->value;
+    }
```

Comment:
Looks like `%settings_by_job_id` could be initialized using `map`. And this is actually `setting_value_by_job_id_by_key`. Not sure whether it make sense to have such a long name but `$settings_by_job_id{TEST_SUITE_DESCRIPTION}` looks weird considering `TEST_SUITE_DESCRIPTION` is not a job id.

When we decide to follow my suggestion from the other comment this could be much simpler anyways.